chainntnfs/btcdnotify/btcd.go

```diff
@@ -528,33 +598,78 @@ func (b *BtcdNotifier) RegisterSpendNtfn(outpoint *wire.OutPoint,
 	// is registered, the output hasn't already been spent. If the output
 	// is no longer in the UTXO set, the chain will be rescanned from the point
 	// where the output was added. The rescan will dispatch the notification.
-	txout, err := b.chainConn.GetTxOut(&outpoint.Hash, outpoint.Index, true)
+	txOut, err := b.chainConn.GetTxOut(&outpoint.Hash, outpoint.Index, true)
 	if err != nil {
 		return nil, err
 	}
 
-	if txout == nil {
-		transaction, err := b.chainConn.GetRawTransactionVerbose(&outpoint.Hash)
+	if txOut == nil {
+		tx, err := b.chainConn.GetRawTransactionVerbose(&outpoint.Hash)
 		if err != nil {
+			// Avoid returning an error if the transaction was not
+			// found.
 			jsonErr, ok := err.(*btcjson.RPCError)
 			if !ok || jsonErr.Code != btcjson.ErrRPCNoTxInfo {
+				return nil, fmt.Errorf("unable to query for "+
+					"txid %v: %v", outpoint.Hash, err)
+			}
+		}
+
+		var blockHash *chainhash.Hash
+		if tx != nil && tx.BlockHash != "" {
+			// If we're able to retrieve a valid block hash from the
+			// transaction, then we'll use it as our rescan starting
+			// point.
+			blockHash, err = chainhash.NewHashFromStr(tx.BlockHash)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			// Otherwise, we'll rely on manually scanning our chain
+			// for the block the transaction was included in,
+			// starting from the earliest height it could have been
+			// included in to the current.
+			_, currentHeight, err := b.chainConn.GetBestBlock()
+			if err != nil {
 				return nil, err
 			}
+
+			targetTxidStr := outpoint.Hash.String()
+		out:
```

Comment:
Similar comment here w.r.t this block not being necessary.

Reply:
Fixed.